src/main/java/uk/gov/hmcts/befta/player/BackEndFunctionalTestScenarioContext.java

```diff
@@ -83,6 +91,14 @@ public String getCurrentScenarioTag() {
             .collect(Collectors.joining(","));
     }
 
+    public String getGuid() {
+        return testData == null ? "" : testData.get_guid_();
+    }
```

Comment:
having both a getGuid and getReference makes it confusing.

Reply:
you case use getId() instead. The reason is: guid in test data objects are globally unique in test data source, hence the name. but for scenarios, ids don't have to be globally unique. Uniqueness is required only within a set of sibling contexts.

Reply:
Replaced `guid` and `reference` properties with a single one called to `contextId` as agreed.